### PR TITLE
Angus/remove splatalogue

### PR DESCRIPTION
**Pull request by veggiesaurus**
Removes all splatalogue functionality from the backend, and also removes use of `curl_easy_escape`. This means we can remove curl as a dependency.

Note that we need to merge the[ frontend PR](https://github.com/CARTAvis/carta-frontend/pull/1876) first or things will break :laughing:

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 * Enhanced image fitting performance by switching the solver from qr to cholesky ([#1114](https://github.com/CARTAvis/carta-backend/pull/1114)).
 * Made HTTP server return a different error code for disabled features ([#1115](https://github.com/CARTAvis/carta-backend/issues/1115)).
+* Removed Splatalogue interfaction from backend codebase and removed dependency onn libcurl ([#994](https://github.com/cartavis/carta-backend/issues/994)).
```
Comment (confluence) [inline CHANGELOG.md]: Typo? Interface or interaction?

**Review by confluence — COMMENTED** [review]
This looks fine. Is there also a protobuf PR to remove these messages?

**Review by confluence — APPROVED** [review]

**Review by confluence — CHANGES_REQUESTED** [review]
Forgot the typo.

File: CHANGELOG.md (hunk 1 above)
Comment (veggiesaurus) [reply CHANGELOG.md]: fixed :)

**Review by confluence — APPROVED** [review]

**Comment by veggiesaurus** [comment]
> This looks fine. Is there also a protobuf PR to remove these messages?

I can't remember how we handled things before. Do we simply just remove the messages and the entries from the enum?

**Comment by acdo2002** [comment]
There are two protobuf related to splatalogue
[73, CARTA.SplataloguePing],
[74, CARTA.SplataloguePong]
There is no ICD tests for these two protobuf.

I have a question, would this PR(remove splatalogue) affect [67, CARTA.SpectralLineRequest] and [68, CARTA.SpectralLineResponse] protobuf? Because we have the ICD tests for these two protobuf.

**Comment by kswang1029** [comment]
> There are two protobuf related to splatalogue [73, CARTA.SplataloguePing], [74, CARTA.SplataloguePong] There is no ICD tests for these two protobuf.
> 
> I have a question, would this PR(remove splatalogue) affect [67, CARTA.SpectralLineRequest] and [68, CARTA.SpectralLineResponse] protobuf? Because we have the ICD tests for these two protobuf.

We can remove all splatalogue related ICD tests after this PR is merged.

**Comment by acdo2002** [comment]
> > There are two protobuf related to splatalogue [73, CARTA.SplataloguePing], [74, CARTA.SplataloguePong] There is no ICD tests for these two protobuf.
> > I have a question, would this PR(remove splatalogue) affect [67, CARTA.SpectralLineRequest] and [68, CARTA.SpectralLineResponse] protobuf? Because we have the ICD tests for these two protobuf.
> 
> We can remove all splatalogue related ICD tests after this PR is merged.

Ok! Then we only need to remove three ICD tests:
SPECTRAL_LINE_QUERY.test.ts
SPECTRAL_LINE_QUERY_INTENSITY_LIMIT.test.ts
SPECTRAL_LINE_QUERY_WIDE_FREQ.test.ts

**Comment by kswang1029** [comment]
@veggiesaurus @confluence no regression is observed via the current set of e2e tests.

**Comment by confluence** [comment]
> I can't remember how we handled things before. Do we simply just remove the messages and the entries from the enum?

Yes, from the history it looks like that's what we do now.

**Review by ajm-ska — APPROVED** [review]
This looks good.

I tried the spectral line query and I feel it is a lot faster now than when it was done by the backend.

**Pull request merged**